foreign_cc/private/cmake_script.bzl

```diff
@@ -2,6 +2,18 @@
 
 load(":cc_toolchain_util.bzl", "absolutize_path_in_str")
 
+def _escape_dquote_bash(text):
+    """ Escape double quotes in flag lists for use in bash strings to be passed to a HEREDOC. """
+
+    # We use a starlark raw string to prevent the need to escape backslashes for starlark as well.
+    return text.replace('"', r'\\\"')
+
+def _escape_dquote_bash_crosstool(text):
+    """ Escape double quotes in flag lists for use in bash strings to be passed to a HEREDOC. """
```

Comment:
Can you update this docstring to comment on why this is different than `_escape_dquote_bash`?